managed/services/agents/agents.go

```diff
@@ -145,11 +145,11 @@ func ensureAuthParams(exporter *models.Agent, params *agentpb.SetStateRequest_Ag
 // getExporterListenAddress returns the appropriate listen address to use for a given exporter.
 func getExporterListenAddress(node *models.Node, exporter *models.Agent) string {
 	switch {
-	case !exporter.PushMetrics && node != nil:
-		return node.Address
 	case exporter.ExposeExporter:
 		return "0.0.0.0"
-	default:
+	case exporter.PushMetrics:
 		return "127.0.0.1"
+	default:
```

Comment:
It seems like this makes the first case redundant.

Reply:
Yeah true - I wanted to keep the case where exposeExporter is enabled at the top so it overrides anything else.